spec/unit/puppet/provider/ntp_config/cisco_nexus_spec.rb

```diff
@@ -0,0 +1,291 @@
+        expect(ntp_config).to receive(:trusted_key_set).with(true, '11').once
+        expect(ntp_config).to receive(:trusted_key_set).with(false, '2').never
+        expect(ntp_config).to receive(:trusted_key_set).with(false, '3').never
+        expect(ntp_config).to receive(:trusted_key_set).with(false, '4').once
+        provider.handle_trusted_keys([2, 3, 5, 10, 11])
+      end
+    end
+    context 'unset is passed' do
+      it 'deletes the old keys' do
+        expect(ntp_config).to receive(:trusted_key).and_return([2, 3]).twice
+        expect(ntp_config).to receive(:trusted_key_set).with(false, '2').once
+        expect(ntp_config).to receive(:trusted_key_set).with(false, '3').once
+        expect(ntp_config).to receive(:trusted_key_set).with(true, anything).never
+        provider.handle_trusted_keys(['unset'])
+      end
+      it 'does not add when key source is empty' do
+        expect(ntp_config).to receive(:trusted_key).once
+        expect(ntp_config).to receive(:trusted_key_set).with(false, anything).never
+        expect(ntp_config).to receive(:trusted_key_set).with(true, anything).never
+        provider.handle_trusted_keys(['unset'])
+      end
+    end
+  end
+
+  validate_should_data = [
+    {
+      desc: '`name` is not `default`',
+      issue: 'raise an error',
+      expect: it { expect { provider.validate_should(name: 'foo') }.to raise_error Puppet::ResourceError, 'Invalid name, `name` must be `default`' }
+    },
+    {
+      desc: '`name` is default',
+      issue: 'not raise an error',
+      expect: it { expect { provider.validate_should(name: 'default') }.not_to raise_error }
+    },
+    {
+      desc: '`source interface` contains spaces',
+      issue: 'raise an error',
+      expect: it {
+        expect { provider.validate_should(name: 'default', source_interface: 'space foo') }.to raise_error Puppet::ResourceError,
+                                                                                                           'Invalid source interface, `source_interface` must not contain any spaces'
+      }
+    },
+    {
+      desc: '`source interface` contains uppercase characters',
+      issue: 'raise an error',
+      expect: it {
+        expect { provider.validate_should(name: 'default', source_interface: 'spaceFoo') }.to raise_error Puppet::ResourceError,
+                                                                                                          'Invalid source interface, `source_interface` must not contain any uppercase characters'
+      }
+    },
+    {
+      desc: '`should` is valid',
+      issue: 'not raise an error',
+      expect: it { expect { provider.validate_should(name: 'default', trusted_key: [1, 2], source_interface: '7') }.not_to raise_error }
+    },
+  ]
+  describe '#validate_should' do
+    validate_should_data.each do |test|
+      context "#{test[:desc]}" do
+        it "#{test[:issue]}" do
+          test[:expect]
```

Comment:
Please unroll this into separate examples. This is not only unnecessary complexity, it is also technically wrong, as it is nesting `it`s.